### ISSUE TITLE
Why the function returns null when second invoking on IE browser only [Dart2JS]

### ISSUE THREAD

**Issue by jumperchen** [dart-lang/sdk#36206]
Dart version: 2.1.0
Browser: IE 11 (version: 11.648.17134.0)
```
import 'dart:math';
getRandom() {
  var rand;
  print('before try');
  try {
    rand = new Random.secure();
  } catch (e) {
    rand = new Random();
  }
  print('after try $rand');
  return rand;
}
main() {
  print('result: ${getRandom() == null}');
  print('result: ${getRandom() == null}');
}
```

The result:
```
before try
after try Instance of 'minified:bg'
result: false
before try
after try null
result: true

```

**Comment by lrhn** [comment]
I'll dare a guess: The `Random.secure` constructor reads its result from a (lazy) static variable with an initializer which throws. This throw is caught the first time. The second time such a lazy variable is read, after a failed initialization, it evaluates to `null`.

So, the implementation should probably make sure to throw each time `Random.secure` is called, not just the first time.

**Comment by sigmundch** [comment]
Not sure why I can't see @lrhn reponse here in the bug, but I got a notification email with his clever insight:

> I'll dare a guess: The Random.secure constructor reads its result from a (lazy) static variable with an initializer which throws. This throw is caught the first time. The second time such a lazy variable is read, after a failed initialization, it evaluates to null.

> So, the implementation should probably make sure to throw each time Random.secure is called, not just the first time.


@rakudrama is working on a CL to fix how Random.secure is written to ensure that the error is thrown every time.

**Comment by kevmoo** [comment]
https://dart-review.googlesource.com/c/sdk/+/96984

**Comment by rakudrama** [comment]
Fixed at 9b185051a956f9e92530e32b8c7ef2b69d38529b

@jumperchen This will take some time to make its way into a released SDK. In the meantime, you can work around the bug by arranging to call `getRandom()` just once, something like:

```dart
Random _random;
Random myGetRandom() => _random ??= getRandom();
```
Under the covers, `Random()` and `Random.secure()` each return a singleton in the js_runtime implementation, so this will not change the behaviour.

**Comment by jumperchen** [comment]
@rakudrama Okay, thanks for the quick fix.
